addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -40,6 +41,27 @@ options.UserValueWidget.include({
     },
 });
 
+Many2oneUserValueWidget.include({
+    /**
+     * @override
+     */
+    async _getSearchDomain() {
+        // We add the current website id to the domain if the model has a
+        // website_id field. Note that the rpcs are cached but we could get all
+        // models that have a website_id field in a single rpc (TODO).
+        const websiteIdField = await this._rpc({
+            model: this.options.model,
+            method: "fields_get",
+            args: [["website_id"]],
+        });
+        const modelHasWebsiteId = !!websiteIdField["website_id"];
+        if (modelHasWebsiteId && !this.options.domain.find(arr => arr[0] === "website_id")) {
+            this.options.domain = this.options.domain.concat(wUtils.websiteDomain(this));
```

Comment:
Don't concat domains like that, we need to use utils, both in JS and in PYTHON.
Luckily, the methods are named the same in both languages.
I'll let you grep `.OR(` and `.AND(` :+1: